huggingface_pipelines/text.py

```diff
@@ -0,0 +1,355 @@
+        """
+        return replace(self, **overwrites)
+
+
+class HFEmbeddingToTextPipeline(Pipeline):
+    def __init__(self, config: EmbeddingToTextPipelineConfig):
+        super().__init__(config)
+        logger.info("Initializing embedding to text model...")
+        self.config = config
+        self.t2t_model = EmbeddingToTextModelPipeline(
+            decoder=self.config.decoder_model,
+            tokenizer=self.config.decoder_model,
+            device=self.config.device
+        )
+        logger.info("Model initialized.")
+
+    def process_batch(self, batch: Dict[str, Any]) -> Dict[str, List[str]]:
+        """
+        Process a batch of embeddings and convert them to text.
+        Args:
+            batch (Dict[str, Any]): Input batch containing embeddings.
+        Returns:
+            Dict[str, List[str]]: Processed batch with decoded texts.
+        """
+        for column in self.config.columns:
+            embeddings = batch[column]
+            assert all(isinstance(item, list) for item in embeddings), \
+                f"Column {column} must contain only lists of embeddings, not individual embeddings."
+            all_embeddings = np.vstack([np.array(embed)
+                                       for item in embeddings for embed in item])
+            all_decoded_texts = self.decode_embeddings(all_embeddings)
+
+            reconstructed_texts = []
+            start_idx = 0
+            for item in embeddings:
+                end_idx = start_idx + len(item)
+                reconstructed_texts.append(
+                    all_decoded_texts[start_idx:end_idx])
+                start_idx = end_idx
+
+            batch[f"{column}_{self.config.output_column_suffix}"] = reconstructed_texts
+            logger.info(
+                f" {column} column reconstructed:  {batch[column][:5]}")
+        return batch
+
+    def decode_embeddings(self, embeddings: np.ndarray) -> List[str]:
+        """
+        Decode a batch of embeddings into text.
+        Args:
+            embeddings (np.ndarray): Array of embeddings to decode.
+        Returns:
+            List[str]: List of decoded texts.
+        Raises:
+            Exception: If there's an error during decoding.
+        """
+        try:
+            logger.info(f"Decoding {len(embeddings)} embeddings...")
+            embeddings_tensor = torch.from_numpy(embeddings).float()
+
+            decoded_texts = []
+            for i in range(0, len(embeddings), self.config.batch_size):
+                batch_embeddings = embeddings_tensor[i:i +
+                                                     self.config.batch_size]
+                batch_decoded = self.t2t_model.predict(
+                    batch_embeddings,
+                    target_lang=self.config.target_lang,
+                    batch_size=self.config.batch_size
+                )
+                decoded_texts.extend(batch_decoded)
```

Comment:
Maybe I miss something here: You flatten the batch of decoded texts, just to reconstruct them later ? Why don't we keep the list of baches here already ?